test/accessibility/apps/management.ts

```diff
@@ -43,6 +44,25 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       await a11y.testAppSnapshot();
     });
 
+    it('Index pattern field editor', async () => {
+      await PageObjects.settings.clickAddField();
+      await a11y.testAppSnapshot();
+
+      // make sure all settings are expanded before taking a snapshot
+      await PageObjects.settings.setFieldName('test');
+      await PageObjects.settings.setFieldType('Keyword');
+      await PageObjects.settings.setFieldScript("emit('hello world')");
+      await PageObjects.settings.toggleRow('formatRow');
+      await PageObjects.settings.setFieldFormat('string');
+      await PageObjects.settings.toggleRow('customLabelRow');
+      await PageObjects.settings.setCustomLabel('custom label');
+      await testSubjects.click('toggleAdvancedSetting');
+
+      await a11y.testAppSnapshot();
```

Comment:
Can we bring this out into its own test? There should only be one `testAppSnapshot()` per `it()` test block.

It's not that it doesn't work like you have (it will); it makes it difficult to debug though. Because the failures will callout which test failed ("Index pattern field editor", in this case) and will give you DOM information about the failure (e.g., "<input id='foo' /> doesn't have a label") and then a stacktrace of, mostly, axe internals which isn't incredibly helpful. So you're left no knowing which snapshot within a test is causing this issue.

So only doing 1 per test avoid a debugging step.

(I realize this isn't documented anywhere and that's my bad. I gotta find the right place to add it. If you have suggestions on where to put it, I'm all ears!)

Reply:
split into two 👍